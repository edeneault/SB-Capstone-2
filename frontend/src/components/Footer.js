import React from "react";
import { Container, Row, Col } from "react-bootstrap";
import { Link } from "react-router-dom";

const Footer = () => {
  return (
    <footer>
      <Container>
        <Row>
          <Col className='text-center py-1'>
            <Link to={{ pathname: "https://www.facebook.com" }} target='_blank'>
              <i className='fab fa-facebook px-4 text-dark '></i>
            </Link>
            <Link
              to={{ pathname: "https://www.instagram.com" }}
              target='_blank'
            >
              <i className='fab fa-instagram px-4 text-dark'></i>
            </Link>
            <Link to={{ pathname: "https://www.linkedin.com" }} target='_blank'>
              <i className='fab fa-linkedin-in px-4 text-dark'></i>
            </Link>
            <Link to={{ pathname: "https://www.tiktok.com" }} target='_blank'>
              <i className='fab fa-tiktok px-4 text-dark'></i>
            </Link>
          </Col>
        </Row>
        <Row>
          <Col className='text-center py-1'>
            Copyright &copy; Impresso Espresso
          </Col>
        </Row>
        <Row>
          <Col className='text-center disclaimer'>
            Web Application built using MERN stack as a Capstone Project. NOT A
            REAL STORE.
          </Col>
        </Row>
      </Container>
    </footer>
  );
};

export default Footer;
