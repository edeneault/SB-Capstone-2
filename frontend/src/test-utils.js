import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
