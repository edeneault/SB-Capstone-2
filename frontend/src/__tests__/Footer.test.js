import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { initialState, render } from "../test-utils";

import Footer from "../components/Footer";

const mockStore = configureMockStore([thunk]);

test("renders Footer component without crashing", () => {
  const store = mockStore(initialState);

  const { getByText } = render(
    <Provider store={store}>
      <Footer />
    </Provider>,
  );

  expect(getByText("Copyright Â© Impresso Espresso")).toBeInTheDocument();
});

test("matches snapshot", function () {
  const store = mockStore(initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <Footer />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
