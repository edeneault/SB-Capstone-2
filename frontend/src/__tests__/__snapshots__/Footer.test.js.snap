// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <footer>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="text-center py-1 col"
        >
          <a
            href="https://www.facebook.com"
            target="_blank"
          >
            <i
              class="fab fa-facebook px-4 text-dark "
            />
          </a>
          <a
            href="https://www.instagram.com"
            target="_blank"
          >
            <i
              class="fab fa-instagram px-4 text-dark"
            />
          </a>
          <a
            href="https://www.linkedin.com"
            target="_blank"
          >
            <i
              class="fab fa-linkedin-in px-4 text-dark"
            />
          </a>
          <a
            href="https://www.tiktok.com"
            target="_blank"
          >
            <i
              class="fab fa-tiktok px-4 text-dark"
            />
          </a>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="text-center py-1 col"
        >
          Copyright Â© Impresso Espresso
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="text-center disclaimer col"
        >
          Full Stack Application built using MERN stack as a Capstone Project. NOT A REAL STORE.
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
